frontend/static/js/app.js

```diff
@@ -138,9 +139,9 @@ Vue.component("correction-view", {
         this.errors.push(err.message);
       }
     },
-    toggleCommentBox: function () {
-      this.hasCheckedComment = !this.hasCheckedComment;
-    },
+    isChecked: function () {

```

Comment:
Can we avoid writing this handler? It is confusing to have both checked member and a isChecked handler. Also as we discussed earlier the component shouldn't know what UI control is used on the UI i.e. this file shouldn't have a word "Checked" anywhere because we don't need it to know that we're using a checkbox on UI.

We can watch for changes to our data member but if that is too much of a trouble, we can simply use the comment or not based on 'useComment' being set to true or false.

Reply:
I will do the changes. 

What is the problem letting code know what we are using in the UI? Just curious!

Reply:
We want separation of concerns. 
- A person looking at the code, shouldn't have to also look at the UI to understand what is going on. 
- If we change the UI, the code shouldn't have to change as well ideally
- The point of using Vue and binding is to not have our code deal with UI and simply focus on business logic

Using and operator will add a boolean to the comment. You can either add a method getComment or you can watch the value of useComment to reset the comment on the model.

Reply:
@hasankhan Understood, thanks. Will add the changes accordingly.